### PR TITLE
Adding support for TLS 1.1 and 1.2

### PR DESCRIPTION
**Pull request by bhariharan**

### CODE REVIEW

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java [hunk 1]
```diff
@@ -206,7 +208,7 @@ public Execution(HttpResponse response) throws IllegalStateException, IOExceptio
      */
     @TargetApi(19)
     public Execution doPost(Map<String, String> headers, URI uri, HttpEntity requestEntity) throws IOException {
-    	final HttpURLConnection httpConn = createHttpUrlConnection(uri, HttpPost.METHOD_NAME);
+    	final HttpsURLConnection httpConn = createHttpUrlConnection(uri, HttpPost.METHOD_NAME);
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java]: Need to explicitly use ```HttpsURLConnection``` for ```SSL```.

File: libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java [hunk 2]
```diff
@@ -406,10 +408,17 @@ private HttpURLConnection createHttpUrlConnection(URI uri, String requestMethod)
     				url = new URL(urlString);
     				requestMethod = HttpPost.METHOD_NAME;
     			}
-    			httpConn = (HttpURLConnection) url.openConnection();
+    			httpConn = (HttpsURLConnection) url.openConnection();
     			httpConn.setRequestMethod(requestMethod);
     			httpConn.setRequestProperty(USER_AGENT, userAgent);
-    		}
+                try {
+                    httpConn.setSSLSocketFactory(new SalesforceTLSSocketFactory());
```
Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java]: Sets the custom socket factory.

Comment (bhariharan) [inline libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/HttpAccess.java]: Maybe I should do this only for pre-Lollipop here?

**Comment by kchitalia** [comment]
:+1:

**Comment by wmathurin** [comment]
:+1:

**Pull request merged**